woocommerce/woocommerce-gutenberg-products-block
# Store Customization > Fetch product images from the Pexels API

## What

Update the image provider for the Customize Your Store experience, migrating from Verticals to Pexels. Additional changes made on this PR also include:

- Performance optimization for the creation of dummy products and assignment of images.
- Ensure the `'woocommerce_blocks_allow_ai_connection'` option is updated to true whenever the store owner updates their business description (either via the onboarding flow or the store admin area). The option that stores the business description is available within WooExpress exclusively. The `'woocommerce_blocks_allow_ai_connection'` option will be updated only if the site has an active connection with JetPack (therefore they accepted the terms and conditions).
- Updates the Prompts used for generating AI content for Products and Patterns. Note: there's an improvement to the AI-assigned images based on the changes made here, but additional fine-tuning is planned as part of a separate PR.

### Implementation details

Instead of making requests directly to the Pexels API, we are now relying on WP.com's endpoint (`'/wpcom/v2/external-media/list/pexels'`) as a middleman to ensure there won't be any restrictions regarding rate limiting. JetPack is a dependency for this connection as their built-in client for this endpoint is being used to power the requests. 

Fixes #11461

## Why

Given several restrictions and limitations we have found with the Verticals API, including low availability of images and incorrect classification of images, this update ensures the assignment of images to patterns and products by AI is consistent and more accurate.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

### Initial Setup

- Request access to https://woo-delightfully-luminous-kryptonite.wpcomstaging.com/ if you still don't have it.

Or alternatively, create a new WordPress install from scratch (make sure the install is a Jurassic Ninja install or equivalent. Otherwise you won't be able to connect with Jetpack to test this PR).

- Install and activate [this version](https://github.com/woocommerce/woocommerce/releases/tag/untagged-eb8de7515421456195de) of WooCommerce.
- Install and activate WooCommerce Blocks from the following zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/13174321/woocommerce-gutenberg-products-block.zip) 
- Install and activate Jetpack.
- In your wp-admin, you should see the following section for Jetpack:

<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://user-images.githubusercontent.com/15730971/267093030-fc804759-f724-4fa2-829d-7bfd441962ec.png">

- Click on "Set up Jetpack" and make sure the connection is successful:

<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://user-images.githubusercontent.com/15730971/267093371-661ee4c5-a0ac-40fd-9e5e-835a0e4c1cb7.png">

- Install and activate Woo AI (the plugin is available [within the monorepo](https://github.com/woocommerce/woocommerce/tree/trunk/plugins))
- Install and activate WooCommerce Beta Tester (the plugin is available [within the monorepo](https://github.com/woocommerce/woocommerce/tree/trunk/plugins))
- Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag:

<img width="1233" alt="Screenshot 2023-10-22 at 10 32 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/816867c6-34e5-40a1-a87d-f4a7cda46429">

### Test the AI-managed images

1. Head over to Home > Customize your Store:
 
<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/b5d34131-3edc-47ed-a8e4-8e2775ca1bbd">

2. Click on design with AI.
3. Type a description for your business.
4. Make sure the AI content is properly generated and images are assigned to patterns and products based on the description provided.

### Known limitations: 

* In a few cases, you may notice that new images are not assigned to a product or pattern in particular. This is still under investigation, and a separate PR will be opened to address this limitation.
* Sometimes, depending on the store description, images are not tightly associated with the expected content: more fine-tuning is happening in upcoming PRs.

* [] Should be tested by the development team exclusively 

## Screenshots or screencast




https://github.com/woocommerce/woocommerce-blocks/assets/15730971/2ee290e8-d6e9-49b0-98ee-e5b59f44a427



## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Update the image provider for the Customize Your Store experience, migrating from Verticals to Pexels.
